external/storm-redis/src/main/java/org/apache/storm/redis/bolt/RedisFilterBolt.java

```
@@ -0,0 +1,146 @@
+ */
+public class RedisFilterBolt extends AbstractRedisBolt {
+    private final RedisFilterMapper filterMapper;
+    private final RedisDataTypeDescription.RedisDataType dataType;
+    private final String additionalKey;
+
+    /**
+     * Constructor for single Redis environment (JedisPool)
+     * @param config configuration for initializing JedisPool
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisPoolConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * Constructor for Redis Cluster environment (JedisCluster)
+     * @param config configuration for initializing JedisCluster
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisClusterConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(Tuple input) {
+        String key = filterMapper.getKeyFromTuple(input);
+
+        boolean found;
+        JedisCommands jedisCommand = null;
+        try {
+            jedisCommand = getInstance();
+
+            switch (dataType) {
+                case STRING:
+                    found = jedisCommand.exists(key);
+                    break;
+
+                case SET:
+                    if (additionalKey == null) {
```

since data type is known at the time of construction, can this check be moved to constructor itself? Failing fast is preferred.